### PR TITLE
Check for existence of claimed time bounds variable

### LINKED ISSUE #43: is_multi_year_mean fails when time:bounds variable does not exist
This likely is due to another method failing by not checking whether the variable named in `time:bounds` actually exists. Fix that.

### PR DESCRIPTION
**Pull request by rod-glover**
Resolves #43

### CODE REVIEW

File: nchelpers/__init__.py [hunk 1]
```diff
@@ -982,10 +982,14 @@ def time_bounds_var_name(self):
...
         # If time:bounds attribute exists, use it.
         try:
-            return time_var.bounds
+            var_name = time_var.bounds
+            if hasattr(self.variables, var_name):
+                return var_name
         except AttributeError:
-            if self._cf_dataset_options['strict_metadata']:
-                return None
+            pass
```
Comment (jameshiebert) [inline nchelpers/__init__.py]: What happens when this falls through?

File: nchelpers/__init__.py [hunk 2]
```diff
@@ -1059,9 +1063,12 @@ def multi_year_bounds(time_bounds, scale):
...
             time_bounds_name = time_var.bounds
-            time_bounds = self.variables[time_bounds_name]
-            if multi_year_bounds(time_bounds, scale):
-                return time_bounds_name
+            try:
+                time_bounds = self.variables[time_bounds_name]
+                if multi_year_bounds(time_bounds, scale):
+                    return time_bounds_name
+            except KeyError:
+                pass  # claimed time bounds variable does not exist
```
Comment (jameshiebert) [inline nchelpers/__init__.py]: Likewise

File: nchelpers/__init__.py (hunk 1 above)
Comment (rod-glover) [reply nchelpers/__init__.py]: It goes on to the next rule/heuristic. That's the structure of this method: a succession of attempts, which if they fail, fall through to the next attempt. This one is perhaps a little tricky: `AttributeError` is raised if there is no bounds attribute on `time_var`; if there is, then the existence of the claimed variable is tested. Failure on either test falls through to the next step.

File: nchelpers/__init__.py (hunk 2 above)
Comment (rod-glover) [reply nchelpers/__init__.py]: Again, this is a succession of rules/heuristics, which if they succeed return a value, if not they fall through to the next rule/heuristic.

**Comment by jameshiebert** [comment]
OK, WFM.

File: nchelpers/__init__.py (hunk 2 above)
Comment (rod-glover) [reply nchelpers/__init__.py]: And similarly a bit tricky ... KeyError raised if the claimed bounds variable doesn't exist (fall through). If not, test whether the existing variable's values look plausible. If not, fall through.

**Pull request merged**